plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFailureRecovery.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.Session;
+import io.trino.testing.AbstractTestFailureRecovery;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class TestHiveFailureRecovery
+        extends AbstractTestFailureRecovery
+{
+    @Override
+    protected QueryRunner createQueryRunner(List<TpchTable<?>> requiredTpchTables, Map<String, String> configProperties, Map<String, String> coordinatorProperties)
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setInitialTables(requiredTpchTables)
+                .setCoordinatorProperties(coordinatorProperties)
+                .setExtraProperties(configProperties)
+                .build();
+    }
+
+    @Override
+    @Test(enabled = false)
```

Comment:
Instead using `enabled = false` change body to:

```java
        assertThatThrownBy(super::testCreateTable)
                .hasMessageContaining("expected failure message");
        throw new SkipException("create table is not atomic at the moment");
```

This way we will remember that test needs to be enabled when we fix Hive connector.

Reply:
(same for others)

Reply:
Oh - actually test pass for Hive. Should it?

Reply:
> This way we will remember that test needs to be enabled when we fix Hive connector.

Thanks a great technique, thanks for the suggestion.

> Oh - actually test pass for Hive. Should it?

It looks like inserts into a new partition work now, as for non S3 writes move is performed

Reply:
If they fail often enough you can use `assertEventually`

Reply:
The failures could be rare (it only fails if one of the tasks managed to successfully commit some data before failure)